nodejs-aws-shop-infrastructure/lib/nodejs-aws-shop-infrastructure-stack.ts

```diff
@@ -0,0 +1,42 @@
+import { Stack, StackProps, RemovalPolicy } from "aws-cdk-lib";
+import { Distribution, ViewerProtocolPolicy } from "aws-cdk-lib/aws-cloudfront";
+import { S3BucketOrigin } from "aws-cdk-lib/aws-cloudfront-origins";
+import { Bucket, BlockPublicAccess } from "aws-cdk-lib/aws-s3";
+import { BucketDeployment, Source } from "aws-cdk-lib/aws-s3-deployment";
+import { Construct } from "constructs";
+import path = require("path");
+
+export class NodejsAwsShopInfrastructureStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const bucket = new Bucket(this, "NodejsAwsShopInfrastructureBucket", {
+      bucketName: "nodejs-aws-shop-infrastructure-s3",
+      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+      websiteIndexDocument: "index.html",
+      websiteErrorDocument: "index.html",
```

Comment:
It's a bit surprising that you have static website hosting enabled with origin access control used. Because "OAC cannot be used with S3 static website endpoints, only S3 buckets".

Reply:
done, thanks for the comment